src/argilla/client/feedback/metrics/annotator_metrics.py

```diff
@@ -0,0 +1,251 @@
+    RatingQuestion,
+    TextQuestion,
+)
+from argilla.utils.dependency import requires_dependencies
+
+if TYPE_CHECKING:
+    from argilla.client.feedback.dataset import FeedbackDataset
+
+import evaluate
+import pandas as pd
+from sklearn.metrics import (
+    accuracy_score,
+    confusion_matrix,
+    f1_score,
+    multilabel_confusion_matrix,
+    precision_score,
+    recall_score,
+)
+
+# Type aliases
+Responses = List[Union[float, int, str]]
+Suggestions = Responses
+
+
+class AnnotatorMetric:
+    def __init__(self, dataset: "FeedbackDataset", question_name: str) -> None:
+        self._dataset = dataset
+        self._question_name = question_name
+        self._question_type = type(self._dataset.question_by_name(question_name))
+        if self._question_type in (MultiLabelQuestion, RankingQuestion):
+            raise NotImplementedError(f"No metrics are defined currently for {self._question_type.__name__}")
+        self._allowed_metrics = METRICS_PER_QUESTION[self._question_type]
+
+    def _prepare_responses_and_suggestions(self) -> Tuple[Dict[int, Responses], Suggestions]:
+        hf_dataset = self._dataset.format_as("datasets")
+        # TODO(plaguss): We should expose this functionality more easily for a FeedbackDataset.
+        # Create a dict of responses per user:
+        responses_per_user = defaultdict(list)
+        import operator
+
+        operator.itemgetter
+        if self._question_type != RankingQuestion:
+            for responses_ in hf_dataset[self._question_name]:
+                for response in responses_:
+                    user_id = response["user_id"]
+                    if user_id is None:
+                        raise NotImplementedError(
+                            "In order to use this functionality the records need to be assigned to a user."
+                        )
+                    responses_per_user[user_id].append(response["value"])
+        else:
+            for responses_ in hf_dataset[self._question_name]:
+                for response in responses_:
+                    user_id = response["user_id"]
+                    if user_id is None:
+                        raise NotImplementedError(
+                            "In order to use this functionality the records need to be assigned to a user."
+                        )
+                    responses_per_user[user_id].append(response["value"]["value"])
+
+        suggestions = hf_dataset[f"{self._question_name}-suggestion"]
+        if self._question_type == RankingQuestion:
+            suggestions = [suggestion["value"] for suggestion in suggestions]
+        # TODO(plaguss): Check there are suggestions for every response.
+
+        return responses_per_user, suggestions
+
+    def compute(self, metric_names: Union[str, List[str]], **kwargs) -> float:
+        if isinstance(metric_names, str):
+            metric_names = [metric_names]
+
+        metric_classes = []
+        for metric_name in metric_names:
```

Comment:
maybe we can use something like `any([metric not in self._allowed_metrics for metric in metric_names])`?